src/univariate/continuous/normal.jl

```diff
@@ -150,8 +150,8 @@ function suffstats(::Type{<:Normal}, x::AbstractArray{T}, w::AbstractArray{Float
     n = length(x)
 
     # compute s
-    tw = 0
-    s = 0
+    tw = zero(T)
+    s = zero(T)
```

Comment:
This should take into account both the type of `w[i]` and `x[i]`:
```suggestion
    s = 0.0 * zero(T)
```